lizrd/core/distributed.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+from functools import partial
+
+from torch.distributed.fsdp import MixedPrecision, CPUOffload
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch.nn as nn
+import torch
+
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+
+def custom_auto_wrap_policy(
+    module: nn.Module,
+    recurse: bool,
+    nonwrapped_numel: int,
+    # Additional custom arguments
+    min_num_params: int = int(1e8),
+) -> bool:
+    return nonwrapped_numel >= min_num_params
+
+
+def wrap_in_fsdp(
+    module: nn.Module,
+    rank: Optional[int],
+    param_precision: torch.dtype,
+    cast_inputs: bool,
+    mixed_precision_ignore_classes: list,
+    offload_params: bool,
+    print_model: bool,
+):
+    def _create_single_fsdp_module(module_to_wrap: nn.Module, precision: torch.dtype):
+        return FSDP(
+            module_to_wrap,
+            device_id=rank,
+            mixed_precision=MixedPrecision(
+                param_dtype=precision,
+                reduce_dtype=torch.float32,
+                cast_forward_inputs=cast_inputs,
+                _module_classes_to_ignore=mixed_precision_ignore_classes,
+            ),
+            cpu_offload=CPUOffload(offload_params=offload_params),
+            auto_wrap_policy=partial(custom_auto_wrap_policy, min_num_params=int(1e04)),
```

Comment:
`min_num_params` should not be hardcoded. please add option to change this to argparse

Reply:
done